backend/ohq/permissions.py

```diff
@@ -392,3 +395,66 @@ def has_permission(self, request, view):
         # Head TAs+ can make changes
         if view.action in ["create", "destroy", "update", "partial_update"]:
             return membership.is_leadership
+
+
+class EventPermission(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return self.has_permission(request, view)
+
+    def has_permission(self, request, view):
+        # Anonymous users can't do anything
+        if not request.user.is_authenticated:
+            return False
+
+        if view.action in ["create", "partial_update", "update"]:
+            course_pk = json.loads(request.body)["courseId"]
+            course = Course.objects.get(pk=course_pk)
+            membership = Membership.objects.filter(course=course, user=request.user).first()
```

Comment:
This actually isn't a safe check to make. It assumes that the DRF renderer is the custom camel case one, but if that ever changes this code will throw a KeyError. Instead you can replace this with the following line (which we already use for other permission classes)

```
membership = Membership.objects.get(course=view.kwargs["course_pk"], user=request.user)
````

Reply:
Hi Armaan! Thanks for the speedy reply. Since this course id is posted in the json body, I don't know how to access it using view.kwargs (I tred what you suggested but view.kwargs only access the pk if it is in the url). Do you know what the best way is to grab info from a POST body?